repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/keystonev2/selfvalidation/KeystoneV2SelfValidationTest.groovy

```diff
@@ -79,20 +75,7 @@ class KeystoneV2SelfValidationTest extends ReposeValveTest {
         mc.getHandlings().get(0).getRequest().getHeaders().getFirstValue("x-tenant-name") == "mytenantname"
 
         and: "The X-Authenticated-By headers should be present at the origin service."
-        def foundPassword = false
-        def foundMultiFactor = false
-        for (Header header : mc.getHandlings().get(0).getRequest().getHeaders()) {
-            if ("X-Authenticated-By".equalsIgnoreCase(header.name)) {
-                if ("Password".equalsIgnoreCase(header.value)) {
-                    foundPassword = true
-                }
-                if ("Mfa".equalsIgnoreCase(header.value)) {
-                    foundMultiFactor = true
-                }
-            }
-        }
-        assertTrue(foundPassword)
-        assertTrue(foundMultiFactor)
+        mc.getHandlings().get(0).getRequest().getHeaders().findAll("X-Authenticated-By").sort() == ["MFA", "PASSWORD"]
```

Comment:
That might make it even worse. So what is happening if I write `handler.request` and `request` is a private field? Groovy shouldn't be generating a `getRequest()` method.

```
groovy:000> class Foo {
groovy:001> private String foo = "foo"
groovy:002> String bar = "bar"
groovy:003> }
===> true
groovy:000> new Foo().foo
===> foo
groovy:000> new Foo().bar
===> bar
groovy:000> new Foo().getFoo()
ERROR groovy.lang.MissingMethodException:
No signature of method: Foo.getFoo() is applicable for argument types: () values: []
Possible solutions: getBar(), getAt(java.lang.String), setBar(java.lang.String), getClass(), notify(), every()
groovy:000> new Foo().getBar()
===> bar
groovy:000> new Foo().@foo
===> foo
groovy:000> new Foo().@bar
===> bar
```

and

```
groovy:000> class Foo {
groovy:001> private String privStr = "private"
groovy:002> public String pubStr = "public"
groovy:003> private String getPrivStr() {
groovy:004> println("getPrivStr")
groovy:005> return privStr
groovy:006> }
groovy:007> public String getPubStr() {
groovy:008> println("getPubStr")
groovy:009> return pubStr
groovy:010> }
groovy:011> }
===> true
groovy:000> new Foo().privStr
getPrivStr
===> private
groovy:000> new Foo().getPrivStr()
getPrivStr
===> private
groovy:000> new Foo().@privStr
===> private
groovy:000> new Foo().pubStr
getPubStr
===> public
groovy:000> new Foo().getPubStr()
getPubStr
===> public
groovy:000> new Foo().@pubStr
===> public
```

I'm an old man and I like my explicit method calls.